controller/kubernetes/kubernetes.go

```diff
@@ -524,6 +565,89 @@ func (lbc *loadBalancerController) GetLBConfigs() ([]*config.LoadBalancerConfig,
 	return lbConfigs, nil
 }
 
+func (lbc *loadBalancerController) getProbeConfig(probeStr string, frontEndServices []*config.FrontendService, backendUUIDSvcMap map[string]string) (string, error) {
+	var probeConfig string
+
+	if probeStr != "" {
+		dec := json.NewDecoder(strings.NewReader(probeStr))
+		probeStructs := &ingressHealthChecks{}
+		err := dec.Decode(&probeStructs)
+		if err != nil {
+			return "", fmt.Errorf("Error unmarshaling k8s livenessProbes: %v err: %v", probeStr, err)
+		}
+		if probeStructs != nil {
+			// form the custom config
+			svcBackendNameMap := lbc.createSvcBackendNameMap(frontEndServices, backendUUIDSvcMap)
+
+			if svcBackendNameMap != nil {
+				for _, probeStruct := range probeStructs.LivenessProbes {
+					if backendName, ok := svcBackendNameMap[probeStruct.Probe.Service]; ok {
```

Comment:
Should we log !ok? in WARN. We can do something like:

```
if backendName, ok := svcBackendNameMap[probeStruct.Probe.Service]; !ok {
logrus.WARNf()
continue
}
// the rest of the code below
```
this way we do take care of logging plus avoid placing a big block of code in if block

Reply:
Instead of "logging and continue", I feel we should just return err, since this means the ingress spec is not correct, thoughts?